### PR TITLE
Add NetStandard 1.6 Support

### PR DESCRIPTION
**Pull request by brthor**
NetCore Target framework isn't intended to be used for libraries, so I switched this over to netstandard, allowing greater compatibility for .net core 2.0 and new .net classic apps.

The GetAssembly Code for netcore was not correct, and causing Issue38 test to fail for me. The new code ensures all possible assemblies that are available to the application are searched.

@esskar

### CODE REVIEW

**Comment by esskar** [comment]
thanks, i will look at it soon.

File: src/Serialize.Linq.NetStandard16/Serialize.Linq.NetStandard16.csproj [hunk 1]
```diff
@@ -1,18 +1,15 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp1.1</TargetFramework>
+    <TargetFramework>netstandard1.6</TargetFramework>
     <RootNamespace>Serialize.Linq</RootNamespace>
-    <Version>1.4.0</Version>
+    <Version>1.4.0$(VersionSuffix)</Version>
```
Comment (brthor) [inline src/Serialize.Linq.NetStandard16/Serialize.Linq.NetStandard16.csproj]: necessary for prerelease versioning

File: src/Serialize.Linq.NetStandard16/ExpressionContext.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+using Microsoft.Extensions.DependencyModel;
+
+namespace Serialize.Linq
+{
+    public class ExpressionContext : ExpressionContextBase
+    {
+        protected override IEnumerable<Assembly> GetAssemblies()
+        {
+#if !NETCOREAPP2_0 && !NETCOREAPP1_1 && !NETSTANDARD1_6
+            return AppDomain.CurrentDomain.GetAssemblies();
```
Comment (brthor) [inline src/Serialize.Linq.NetStandard16/ExpressionContext.cs]: Doesn't seem to get all the assemblies in net core

File: src/Serialize.Linq.NetStandard16/ExpressionContext.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#else
+            var depContext = DependencyContext.Default;
```
Comment (brthor) [inline src/Serialize.Linq.NetStandard16/ExpressionContext.cs]: `.deps.json` is a file in every netcore application. It lists the entire dependency closure for the application, and is merged with the currently used framework's `.deps.json` file in the installation directory.

**Comment by brthor** [comment]
Thanks, added a few explanatory comments

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
+.idea
```
Comment (brthor) [inline .gitignore]: Project Rider from Jetbrains, am developing on a Mac

File: src/Serialize.Linq.NetStandard16/Serialize.Linq.NetStandard16.csproj (hunk 1 above)
Comment (brthor) [reply src/Serialize.Linq.NetStandard16/Serialize.Linq.NetStandard16.csproj]: but also versionsuffix is by default empty, so doesn't affect current versioning scheme

File: src/Serialize.Linq.NetStandard16/Serialize.Linq.NetStandard16.csproj [hunk 5]
```diff
@@ -1,18 +1,15 @@
...
     <Company />
-    <Product>Serialize.Linq.NetCore11</Product>
+    <Product>Serialize.Linq.NetStandard16</Product>
```
Comment (brthor) [inline src/Serialize.Linq.NetStandard16/Serialize.Linq.NetStandard16.csproj]: I think `Serialize.Linq.NetStandard` would be a better name, leaving the version info out

File: src/Serialize.Linq.NetStandard16/Serialize.Linq.NetStandard16.csproj [hunk 6]
```diff
@@ -1,18 +1,15 @@
...
     <Authors />
     <Description>C# library to serialize LINQ expressions</Description>
     <Copyright>Copyright © 2012-2017</Copyright>
     <PackageProjectUrl>https://github.com/esskar/Serialize.Linq</PackageProjectUrl>
-    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
```
Comment (brthor) [inline src/Serialize.Linq.NetStandard16/Serialize.Linq.NetStandard16.csproj]: Hey I can put these back, but they break the build on OSX, I'll add a platform condition for windows.

**Comment by brthor** [comment]
This has been sitting here months. Don't want to rebase, will maintain functionality in a fork.

**Pull request closed without merging**